src/eth1/impl/ethers.ts

```diff
@@ -154,4 +157,39 @@ export class EthersEth1Notifier extends EventEmitter implements Eth1Notifier {
     return !(!code || code === '0x');
   }
 
+  private async isAfterEth2Genesis(): Promise<boolean> {
+    const logs = await this.getContractPastLogs([this.contract.interface.events.Eth2Genesis.topic]);
+    return logs.length > 0;
+  }
+
+  private async getPastDepositEvents(): Promise<Deposit[]> {
+    const logs = await this.getContractPastLogs([this.contract.interface.events.Deposit.topic]);
+    return logs.map((log) => {
+      const logDescription = this.contract.interface.parseLog(log);
+      return this.createDeposit(logDescription.values.dataHex, logDescription.values.indexHex);
+    });
+  }
+
+  private async getContractPastLogs(topics: string[]): Promise<Log[]> {
+    const filter = {
+      fromBlock: this.opts.depositContract.deployedAt,
+      toBlock: this.genesisBlockHash,
```

Comment:
I think these should be allowed to be changed, we may want to look at past logs from after deployedAt, and after genesisBlockHash

Reply:
But we should obtain them with syncing right?

Reply:
@wemeetagain Am I assuming this correctly? :)

Reply:
I think we still want to collect deposits from the eth1 chain directly as an additional data source for our validators. We'll be getting 'pending' (not in chain yet) deposits from the eth1 chain and maybe from other peers. What we have right now roughly is set up to do that (the event handler on L63), but if we don't sweep the deposit logs until "now", there's still the potential of missing pending deposits that are in eth1 blocks before "now", when the deposit event handler starts.

Reply:
Oh yeah, BeaconBlock doesn't contain DepositData array. I will expose method getPastDeposits(fromblock, toBlock?) right?